common/app/common/pages.cljc

```diff
@@ -350,6 +356,15 @@
 (defmethod change-spec :del-media [_]
   (s/keys :req-un [::id]))
 
+(defmethod change-spec :add-component [_]
+  (s/keys :req-un [::id ::name ::new-shapes]))
```

Comment:
`::new-shapes` is undefined, and I think in this case it can be called `:internal.changes.add-component/objects` (?)

Reply:
Changed. I prefer to call it "shapes" because it's ot an indexed map, but a vector.

Reply:
;)